### ISSUE TITLE
[darwin]go: cannot install cross-compiled binaries when GOBIN is set

### ISSUE THREAD

**Issue by lowang-bh** [volcano-sh/volcano#2894]
Why GOBIN is set? It will also work via adding gobin dir to PATH env, I think.

**What happened**:
```sh
go: cannot install cross-compiled binaries when GOBIN is set
```
![image](https://github.com/volcano-sh/volcano/assets/21003791/392851c4-74a6-48b1-a1f9-6105f8144df9)

**What you expected to happen**:

Remove the GOBIN in makefile, and missed package can be installed when ever on linux or macos;

**How to reproduce it (as minimally and precisely as possible)**:
1. make controller-gen (on macos)

**Anything else we need to know?**:

1. GOBIN is set in makefile

```sh
ifeq (,$(shell go env GOBIN))
GOBIN=$(shell go env GOPATH)/bin
else
GOBIN=$(shell go env GOBIN)
endif
```
2. go install works if no GOBIN env
![image](https://github.com/volcano-sh/volcano/assets/21003791/678c39ac-9e81-4bd6-ae04-bbc6a5dd8858)

**Environment**:
- Volcano Version: master branch
- Kubernetes version (use `kubectl version`):
- Cloud provider or hardware configuration:
- OS (e.g. from /etc/os-release):
- Kernel (e.g. `uname -a`):
- Install tools:
- Others:

**Comment by hwdef** [comment]
It is reasonable.

**Comment by lowang-bh** [comment]
> It is reasonable.

What's your meaning? I think it's better to leave GOBIN unset to be more compareable.

**Comment by hwdef** [comment]
> > It is reasonable.
> 
> What's your meaning? I think it's better to leave GOBIN unset to be more compatible.

I agree with you.

**Comment by xiao-jay** [comment]
In mac m1 
```
➜  volcano git:(remove-vendor) ✗ make controller-gen
go: creating new go.mod: module tmp
go: cannot install cross-compiled binaries when GOBIN is set
make: *** [controller-gen] Error 1
```

when I delete set GOBIN in makefile
```
// delete 
ifeq (,$(shell go env GOBIN))
GOBIN=$(shell go env GOPATH)/bin
else
GOBIN=$(shell go env GOBIN)
endif
```
result :
```
➜  volcano git:(remove-vendor) make controller-gen
go: creating new go.mod: module tmp

```

@lowang-bh I agree with you.

**Comment by xiao-jay** [comment]
@lowang-bh  But i found when I delete the GOBIN,running `make manifests` will error
```
➜  volcano git:(remove-vendor) ✗ make manifests
go: creating new go.mod: module tmp
/controller-gen "crd:crdVersions=v1,generateEmbeddedObjectMeta=true" paths="./vendor/volcano.sh/apis/pkg/apis/scheduling/v1beta1;./vendor/volcano.sh/apis/pkg/apis/batch/v1alpha1;./vendor/volcano.sh/apis/pkg/apis/bus/v1alpha1;./vendor/volcano.sh/apis/pkg/apis/nodeinfo/v1alpha1" output:crd:artifacts:config=config/crd/bases
/bin/sh: /controller-gen: No such file or directory
make: *** [manifests] Error 127

```

**Comment by lowang-bh** [comment]
> @lowang-bh But i found when I delete the GOBIN,running `make manifests` will error
> 
> ```
> ➜  volcano git:(remove-vendor) ✗ make manifests
> go: creating new go.mod: module tmp
> /controller-gen "crd:crdVersions=v1,generateEmbeddedObjectMeta=true" paths="./vendor/volcano.sh/apis/pkg/apis/scheduling/v1beta1;./vendor/volcano.sh/apis/pkg/apis/batch/v1alpha1;./vendor/volcano.sh/apis/pkg/apis/bus/v1alpha1;./vendor/volcano.sh/apis/pkg/apis/nodeinfo/v1alpha1" output:crd:artifacts:config=config/crd/bases
> /bin/sh: /controller-gen: No such file or directory
> make: *** [manifests] Error 127
> ```

Have you add go's bin path to os's PATH env?

**Comment by hwdef** [comment]
still need

**Comment by yeahdongcn** [comment]
This could be closed. @lowang-bh

**Comment by lowang-bh** [comment]
/close

fixed by #3506
